manifests/router/connector.pp

```diff
@@ -25,12 +27,25 @@
   Optional[String] $sasl_mech = 'ANONYMOUS',
   Optional[String] $sasl_username = undef,
   Optional[String] $sasl_password = undef,
+  Stdlib::Absolutepath $sasl_password_file = '/etc/qpid-dispatch/connector_pw.conf',
   Optional[Enum['normal', 'inter-router', 'route-container']] $role = undef,
   Optional[String] $ssl_profile = undef,
   Optional[Integer[0]] $idle_timeout = undef,
   String $config_file = $qpid::router::config_file,
 ){
 
+  if defined('$sasl_password') {
+    include qpid::params
+
+    file { $sasl_password_file:
```

Comment:
If there's no password, should this file be ensured absent?

Reply:
Am I correct in understanding that we should never use `qpid::router::connector` multiple times? I don't want to create a situation where connector A creates the file with default path, and connector B has no password and clobbers it.

Reply:
I think that is correct so I went ahead and implemented it.

Reply:
Today, no. But possibly. I have been investigating a performance BZ around multiple connectors and links: https://bugzilla.redhat.com/show_bug.cgi?id=1754894

Reply:
It's a define, so you're right that it could be.